src/crimson/os/seastore/onode_manager/staged-fltree/stages/key_layout.h

```diff
@@ -497,12 +511,16 @@ class key_hobj_t {
     [[maybe_unused]] auto oid_masked = string_view_masked_t::decode(oid, delta);
     // TODO(cross-node string dedup)
     assert(oid_masked.get_type() == string_view_masked_t::Type::STR);
+    std::string key;
+    [[maybe_unused]] auto key_masked = string_view_masked_t::decode(key, delta);
+    assert(key_masked.get_type() == string_view_masked_t::Type::STR);
     snap_t snap;
     ceph::decode(snap, delta);
     gen_t gen;
     ceph::decode(gen, delta);
-    return key_hobj_t(ghobject_t(
-        shard_id_t(shard), pool, crush, nspace, oid, snap, gen));
```

Comment:
It should be fine to adjust the constructor of `ghobject_t` directly. IIUC it is the only user of the specific constructor.